internal/pprocess/pprocess.go

```diff
@@ -0,0 +1,10 @@
+// Package pprocess stands for Periodic Process
+package pprocess
+
+import "context"
+
+// PProcess defined the periodic process interface
+type PProcess interface {
```

Comment:
As a rule of thumb, do not introduce a generic interface for a specific problem unless you have 3 similar problems that could use a generic interface.